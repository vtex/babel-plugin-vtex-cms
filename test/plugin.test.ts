import plugin from '../src';
import { transformSync } from '@babel/core';

const example = `
import React from "react";
import RichText from "vtex.store-components/RichText";

const HelloWorld = () => (
  <div>
    <RichText
      textAlignment="CENTER"
      textPosition="CENTER"
      text="# Hello, World!"
    />
  </div>
);

HelloWorld.displayName = 'hello-world';

HelloWorld.schema = {
  title: 'Hello World',
  properties: {
    text: {
      title: "Text",
      type: "string",
    },
    alignment: {
      title: "Alignment",
      type: "string",
    }
  }
}

export default HelloWorld;
`;

it('works', () => {
  const tranformResult = transformSync(example, {
    plugins: [plugin, '@babel/plugin-syntax-jsx'],
  });
  expect(tranformResult?.code).toMatchSnapshot();
});
