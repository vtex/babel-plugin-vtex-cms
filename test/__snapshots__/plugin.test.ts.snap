// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works 1`] = `
"import React from \\"react\\";
import RichText from \\"vtex.store-components/RichText\\";

const HelloWorld = () => <div>
    <RichText textAlignment=\\"CENTER\\" textPosition=\\"CENTER\\" text=\\"# Hello, World!\\" />
  </div>;

export default HelloWorld;"
`;
